README.md

```diff
@@ -1,19 +1,251 @@
+
+In the case of `get` operations, invalidation messages are not generated and reads are performed on the local map.
+
+An illustration of the above logic is shown below.
+
+![Invalidation with Local Cache Region Factory](images/HZLocalCacheRgnFactory.jpg)
+
+If your operations are mostly reads, then this option gives better performance.
+
+***NOTE:*** *If you use `HazelcastLocalCacheRegionFactory`, you cannot see your maps on [Management Center](https://docs.hazelcast.org/docs/management-center/latest/manual/html/index.html).*
+
+With `HazelcastLocalCacheRegionFactory`, all of the following caches are not distributed and are kept locally in the Hazelcast member.
+
+- Entity Cache
+- Collection Cache
+- Timestamp Cache
+
+Entity and Collection are invalidated on update. When they are updated on a member, an invalidation message is sent to all other members in order to remove the entity from their local cache. When needed, each member reads that data from the underlying DB. 
+
+Timestamp cache is replicated. On every update, a replication message is sent to all the other members.
+
+Eviction support is limited to maximum size of the map (defined by `max-size` configuration element) and TTL only. When maximum size is hit, 20% of the entries will be evicted automatically.
+
+### Configuring Query Cache and Other Settings
+
+- To enable use of query cache:
+
+	```xml
+	<property name="hibernate.cache.use_query_cache">true</property>
+	```
+
+- To force minimal puts into query cache:
+
+	```xml
+	<property name="hibernate.cache.use_minimal_puts">true</property>
+	```
+
+- To avoid `NullPointerException` when you have entities that have composite keys (using `@IdClass`):
+
+    ```xml
+	<property name="hibernate.session_factory_name">yourFactoryName</property>
+	```
+	
+***NOTE:*** *QueryCache is always LOCAL to the member and never distributed across Hazelcast Cluster.*
+
+## Configuring Hazelcast for Hibernate
+
+To configure Hazelcast for Hibernate, put the configuration file named `hazelcast.xml` into the root of your classpath. If Hazelcast cannot find `hazelcast.xml`, then it will use the default configuration.
+
+You can define a custom-named Hazelcast configuration XML file with one of these Hibernate configuration properties. 
+
+```xml
+<property name="hibernate.cache.provider_configuration_file_resource_path">
+  hazelcast-custom-config.xml
+</property>
+```
+
+
+```xml
+<property name="hibernate.cache.hazelcast.configuration_file_path">
+  hazelcast-custom-config.xml
+</property>
+```
+
+If you're using Hazelcast client (`hibernate.cache.hazelcast.use_native_client=true`), you can specify a custom Hazelcast client configuration file by using the same parameters.
+
+Hazelcast creates a separate distributed map for each Hibernate cache region. You can easily configure these regions via Hazelcast map configuration. You can define **backup**, **eviction**, **TTL** and **Near Cache** properties.
```

Comment:
Considering some previously asked SOF questions, I think this part is not clear to users. Would it be better if we add a sample configuration here (also for Local cache config maps) and mention that they should be in server configuration when hazelcast-client is preferred?
e.g.
```xml
<map name="region_name">
        <time-to-live-seconds>0</time-to-live-seconds>
        <eviction size="50" />
</map>
```